src/bitcoin/serialize.h

```diff
@@ -45,6 +46,20 @@ inline constexpr uint64_t MAX_SIZE = 0x02000000;
 struct deserialize_type {};
 inline constexpr deserialize_type deserialize{};
 
+//! Safely convert odd char pointer types to standard ones.
+inline char *CharCast(char *c) {
+    return c;
+}
+inline char *CharCast(uint8_t *c) {
+    return (char *)c;
```

Comment:
A few things:

- C-style casts are not considered idiomatic these days and some compilers warn on them. The nice and ugly `reinterpret_cast` is good here because it really illustrates the evil of what is transpiring with these....
- I have been using `std::byte` lately to represent all byte blobs. I sort of wish the bitcoin serialize stuff just used that instead of `char *` ... although I guess that would make your PR larger, so nevermind.